### PR TITLE
feat: add pagespeed insights link

### PR DESCRIPTION
**Pull request by langswei**
Add link to PageSpeed Insights in URL facet

## Description

Added a link to PageSpeed Insights for each given URL in the URL facet.

## Motivation and Context

Meant to give a clue to users how to diagnose issues and make it faster to do so even if they don't need the clue.

## How Has This Been Tested?

https://pagespeed--helix-website--adobe.aem.live/tools/rum/explorer.html?domain=emigrationbrewing.com&filter=&view=month&domainkey=open&conversion.checkpoint=click

## Screenshots (if appropriate):

Screenshot 2024-06-07 at 3 12 41 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tools/rum/elements/link-facet.js [hunk 1]
```diff
@@ -16,10 +16,13 @@ export default class LinkFacet extends ListFacet {
       u.searchParams.set('proxyurl', labelText);
       return `
       <img loading="lazy" src="${u.href}" title="${labelText}" alt="thumbnail image for ${labelText}" onerror="this.classList.add('broken')">
+      <a href="https://pagespeed.web.dev/analysis?url=${labelText}" target="_new"><img title="PageSpeed Insights" alt="PageSpeed Insights" class="pagespeed" src="/tools/rum/pagespeed_48.png"></a>
```
Comment (maxakuru) [inline tools/rum/elements/link-facet.js:19]:
```suggestion
      <a href="https://pagespeed.web.dev/analysis?url=${encodeURIComponent(labelText)}" target="_new"><img title="PageSpeed Insights" alt="PageSpeed Insights" class="pagespeed" src="/tools/rum/pagespeed_48.png"></a>
```

File: tools/rum/elements/link-facet.js [hunk 2]
```diff
@@ -16,10 +16,13 @@ export default class LinkFacet extends ListFacet {
...
       <a href="${labelText}" target="_new">${labelText}</a>`;
     }
     if (labelText.startsWith('https://') || labelText.startsWith('http://')) {
-      return `<a href="${labelText}" target="_new">${labelText}</a>`;
+      return `
+      <a href="https://pagespeed.web.dev/analysis?url=${labelText}" target="_new"><img title="PageSpeed Insights" alt="PageSpeed Insights" class="pagespeed" src="/tools/rum/pagespeed_48.png"></a>
```
Comment (maxakuru) [inline tools/rum/elements/link-facet.js:24]:
```suggestion
      <a href="https://pagespeed.web.dev/analysis?url=${encodeURIComponent(labelText)}" target="_new"><img title="PageSpeed Insights" alt="PageSpeed Insights" class="pagespeed" src="/tools/rum/pagespeed_48.png"></a>
```

**Review by maxakuru — APPROVED** [review]
🚀

**Comment by trieloff** [comment]
The page speed icon should not be an `img` – we can add that in CSS

**Comment by langswei** [comment]
> The page speed icon should not be an `img` – we can add that in CSS, maybe with something like this https://codepen.io/trieloff-the-flexboxer/pen/qBGXjxx

I probably missed your point on this -- I had only read the initial comment which showed up in email, not the edited one here which is more pure CSS.  Anyways have a look and advise if I should abandon that one.

https://github.com/adobe/helix-website/pull/555

**Pull request closed without merging**